File: Python/0053-e-maximum-subarray.py
"""
Prefix sum
"""
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0
        minSum = 0
        res = float('-inf')
        for i in range(len(nums)):
            sum += nums[i]
            res = max(res, sum - minSum)
            minSum = min(minSum, sum)
        return res

File: leetcode/C++/0053-e-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, minSum = 0, res = INT_MIN;
        for (size_t i = 0; i < nums.size(); i++) {
            sum += nums[i];
            res = max(res, sum - minSum);
            minSum = min(minSum, sum);
        }
        return res;
    }
};